build/bazel/remote/execution/v2/remote_execution.proto

```diff
@@ -433,6 +433,26 @@ message Action {
   // and allows disowning an entire set of ActionResults that might have been
   // poisoned by buggy software or tool failures.
   bytes salt = 9;
+
+  // The optional platform requirements for the execution environment. The
+  // server MAY choose to execute the action on any worker satisfying the
+  // requirements, so the client SHOULD ensure that running the action on any
+  // such worker will have the same result.  A detailed lexicon for this can be
+  // found in the accompanying platform.md.
+  // New in version 2.2: clients capable of using version 2.2 of the
+  // specification with a server that advertises using
+  // [ServerCapabilities][build.bazel.remote.execution.v2.ServerCapabilities]'s
+  // high_api_version as at least 2.2 MUST populate this field instead of that
+  // which was deprecated inside the
+  // [Command][build.bazel.remote.execution.v2.Command].  If the server does
+  // not advertise itself as supporting 2.2 then the client MUST populate the
+  // platform properties inside the
+  // [Command][build.bazel.remote.execution.v2.Command] as before and SHOULD
+  // populate this new member too with the same information to facilitate
+  // upgrading.
+  // Execution services conforming to 2.2 MUST ignore platform properties
```

Comment:
With this requirement in place, how will a REv2.1 client be able to run actions on a REv2.2 cluster? The problem is that there's no way to attach desired protocol versions to individual requests. There is no two-way version negotiation between client and server.

I think all of this is far too exact. Relying on API versions, etc. Isn't it sufficient to just say:
- Clients SHOULD set the platform properties both in Action and Command.
- Servers SHOULD prefer the platform properties in the Action over the ones specified in the Command.
- Command.platform is deprecated and will be removed in REv3.

Reply:
My aim was to reduce the amount of redundant data being sent when the server clearly advertises that it's at 2.2, which it can through capabilities. Totally fine to relax this however.